internal/datastore/memdb/memdb.go

```diff
@@ -108,8 +112,13 @@ func (mdb *memdbDatastore) SnapshotReader(revision datastore.Revision) datastore
 		revIndex = len(mdb.revisions) - 1
```

Comment:
shouldn't this also check that `len(mdb.revisions) > 1` so that revIndex can't go negative?

Reply:
We check the number of revisions above